Model/SourcesRepository.php

```diff
@@ -0,0 +1,154 @@
+
+/**
+ * Class SourcesRepository
+ * @package Ampersand\DisableStockReservation\Model
+ */
+class SourcesRepository
+{
+    /**
+     * @var SourcesFactory
+     */
+    protected $sourcesFactory;
+
+    /**
+     * @var Sources
+     */
+    protected $sourcesResourceModel;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultFactory;
+
+    /**
+     * @var SourcesConverter
+     */
+    private $sourcesConverter;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * SourcesRepository constructor.
+     * @param SourcesFactory $sourcesFactory
+     * @param Sources $sourcesResourceModel
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory
+     * @param SourcesConverter $sourcesConverter
+     * @param CollectionFactory $collectionFactory
+     */
+    public function __construct(
+        SourcesFactory $sourcesFactory,
+        Sources $sourcesResourceModel,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory,
+        SourcesConverter $sourcesConverter,
+        CollectionFactory $collectionFactory
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->sourceSelectionResultFactory = $sourceSelectionResultFactory;
+        $this->sourcesConverter = $sourcesConverter;
+        $this->collectionFactory = $collectionFactory;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws \Exception
+     */
+    public function getByOrderId(string $orderId): SourceModel
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        if (!$sourcesModel->getId()) {
+            throw new NoSuchEntityException(__('Source model with the order ID "%1', $orderId));
+        }
+
+        return $sourcesModel;
+    }
+
+    public function getOrdersSourcesCollection(OrderSearchResultInterface $result)
+    {
+        $resultIds = [];
+        foreach ($result as $resultItem) {
+            $resultIds [] = $resultItem->getEntityId();
+        }
+
+        $ordersSourcesItems = $this->collectionFactory->create()
+            ->addFieldToFilter('order_id', ['in' => $resultIds])
+            ->getItems();
+
+        $orderSources = [];
+        foreach ($ordersSourcesItems as $item) {
+            $orderSources[$item->getOrderId()] = $this->sourcesConverter->
+            convertSourcesArrayToSourceSelectionItems($item->getSources());
+        }
+
+        return $orderSources;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceSelectionResultInterface|null
+     */
+    public function getSourceSelectionResultByOrderId(string $orderId): ?SourceSelectionResultInterface
```

Comment:
What's the purpose of this function? The goal of this PR is to:
- Save the sources to our sources table when an order source is calculated
- Load the sources and set the extension attributes when an order is loaded

This appears to be creating a source results object? This seems like a needless step between the conversion process. Your converter functionality should serialize the data and unserliaze the data into sources. I don't think we need a step to create a results object to simply call `getSourceSelectionItems` against?